grakn-factory/janus-factory/src/main/java/ai/grakn/factory/TxFactoryJanus.java

```diff
@@ -110,24 +130,27 @@ private synchronized JanusGraph newJanusGraph(String name, String address, Prope
     }
 
     private JanusGraph configureGraph(String name, String address, Properties properties, boolean batchLoading){
-        //Load default properties if none provided
-        if(properties == null){
-            properties = new Properties();
-            try (InputStream in = getClass().getResourceAsStream(DEFAULT_CONFIG)) {
-                properties.load(in);
-                in.close();
-            } catch (IOException e) {
-                throw new RuntimeException(ErrorMessage.INVALID_PATH_TO_CONFIG.getMessage(DEFAULT_CONFIG), e);
-            }
-        }
-
-
         JanusGraphFactory.Builder builder = JanusGraphFactory.build().
                 set("storage.hostname", address).
                 set("storage.cassandra.keyspace", name).
                 set("storage.batch-loading", batchLoading);
 
-        properties.forEach((key, value) -> builder.set(key.toString(), value));
+        String storageBackend = "storage.backend";
+
+        //Load Defaults
+        DEFAULT_PROPERTIES.forEach((key, value) -> builder.set(key.toString(), value));
+
+        //Load Passed in properties
+        properties.forEach((key, value) -> {
+
+            //Overwrite storage
+            if(key.equals(storageBackend)){
+                value = storageBackendMapper.get(value);
+            }
+
+            builder.set(key.toString(), value);
+        });
```

Comment:
This kind of indirection is very confusing for both users and developers. However, if we really have to hide cassandra from the config file I suppose it's unavoidable.

Reply:
:(